src/audio/AudioWorld.cpp

```diff
@@ -266,30 +169,21 @@ namespace World
         if (!wav.open() || !wav.read())
             return Handle::SfxHandle::makeInvalidHandle();
 
-        alGenBuffers(1, &snd->m_Handle);
-
-        ALenum error = alGetError();
-        if (error != AL_NO_ERROR)
+        try
         {
-            static bool warned = false;
-            if (!warned)
-            {
-                LogWarn() << "Could not create OpenAL buffer: "
-                          << AudioEngine::getErrorString(error);
-                return Handle::SfxHandle::makeInvalidHandle();
-            }
-            return Handle::SfxHandle::makeInvalidHandle();
+            const std::int16_t* dataPtr = (std::int16_t*)(wav.getData());
```

Comment:
I'd try to avoid old style cast.

Reply:
> I'd try to avoid old style cast.

why?

Reply:
You don't really know what is happening there. And it's quite hard to refactor.